scio-core/src/main/scala/com/spotify/scio/io/PubsubIO.scala

```diff
@@ -210,13 +212,21 @@ private final case class PubsubIOWithAttributes[T: ClassTag: Coder](
       w = w.withTimestampAttribute(timestampAttribute)
     }
     val coder = CoderMaterializer.beam(data.context, Coder[T])
-    data
-      .map { kv =>
-        val payload = CoderUtils.encodeToByteArray(coder, kv._1)
-        val attributes = kv._2.asJava
-        new beam.PubsubMessage(payload, attributes)
-      }
-      .applyInternal(w)
+
+    data.applyInternal(new PTransform[PCollection[WithAttributeMap], PDone]() {
```

Comment:
Just curious, does `data.transform` not work for this case?

Reply:
no :/ since it's a sink transform that returns a `PDone`, whereas `SCollection#transform` expects something a mapping from `SCollection => SCollection`.